src/server/qtquick/wtextureprovider.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2024 Yixue Wang <wangyixue@deepin.org>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+#pragma once
+
+#include <wglobal.h>
+#include <qwtexture.h>
+
+#include <QSGTextureProvider>
+
+WAYLIB_SERVER_BEGIN_NAMESPACE
+class WTextureProvider : public QSGTextureProvider
+{
+    Q_OBJECT
+public:
+    virtual QW_NAMESPACE::QWTexture *qwTexture() const = 0;
```

Comment:
Needs QWBuffer too?

Reply:
For TextureProvider in "woutputviewport.cpp", it doesn't store a copy of the source buffer. I'm not sure whether the source buffer will change or not after the texture is constructed.

Reply:
If one wants a QWBuffer, he can directly draw the texture on a buffer.

Reply:
Or we could ensure that "texture is consistent with buffer" at a higher level.

Reply:
done